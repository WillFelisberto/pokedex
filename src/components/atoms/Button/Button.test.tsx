import { render, fireEvent, screen } from '@/tests/test-utils';

import { Button } from '.';

describe('<Button />', () => {
  it('should render with text', () => {
    const { container } = render(<Button text="Click me" />);
    expect(screen.getByText('Click me')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render with an icon', () => {
    render(<Button text="Launch" icon={<span data-testid="icon">ğŸš€</span>} />);
    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });

  it('should handle onClick event', () => {
    const onClickMock = jest.fn();
    render(<Button text="Click me" onClick={onClickMock} />);

    fireEvent.click(screen.getByText('Click me'));
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('should be disabled when the prop is set', () => {
    const { container } = render(<Button text="Disabled" disabled />);
    expect(container.firstChild).toBeDisabled();
  });

  it('should accept custom classes', () => {
    const { container } = render(<Button text="Styled" className="px-5" />);
    expect(container.firstChild).toHaveClass('px-5');
  });
});
