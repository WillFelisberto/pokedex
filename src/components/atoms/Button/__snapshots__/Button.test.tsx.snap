// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render with text 1`] = `
<div>
  <button
    class="flex items-center justify-center gap-2 px-4 py-2 rounded-lg bg-blue-600 text-white transition disabled:opacity-50 disabled:cursor-not-allowed "
  >
    <span>
      Click me
    </span>
  </button>
</div>
`;
